pages/studentAssessmentExam/studentAssessmentExam.js

```diff
@@ -2,6 +2,7 @@ var ERR = require('async-stacktrace');
 var express = require('express');
 var router = express.Router();
 
+const { checkPasswordOrRedirect } = require('../../middlewares/studentAssessmentAccess');
```

Comment:
I’m not in love with requiring middleware from page routes, but I don’t strongly object either. We could shift this function to `lib/`, but I’m not convinced it would be neater. I wanted to at least raise the question to see whether you had any other ideas!

Reply:
Yeah, I don't love this either, but if we move this to `lib/`, I think we'd have to move much of `studentAssessmentAccess` too (along with the `.ejs` file), which basically feels like we're still importing middleware, just without `/middleware/` in the path 😅 I'm inclined to leave this as-is.

Reply:
Agreed. Let's stick with things as you have them.